### ISSUE TITLE
[functional testing]  Functional tests using GPG fail when run against staging VMs

### ISSUE THREAD

**Issue by zenmonkeykstop** [freedomofpress/securedrop#3687]
## Description

Functional tests using GPG fail when run against staging VMs via the instance_information.json file.

## Steps to Reproduce

1) Bring up Vagrant staging VMs
2) set up test database on app_staging
3) update securedrop/tests/functional/instance_information.json with test account info and information from install_files/ansible_base/{app-source-ths, app-journalist-aths}
4) run functional tests

## Expected Behavior
All tests pass

## Actual Behavior
Tests that include GPG decryption fail - for example:
```
self = <tests.functional.test_submit_and_retrieve_file.TestSubmitAndRetrieveFile object at 0x7faa80da4850>

def test_submit_and_retrieve_happy_path(self):
self._source_visits_source_homepage()
self._source_chooses_to_submit_documents()
self._source_continues_to_submit_page()
self._source_submits_a_file()
self._source_logs_out()
self.swap_drivers() # To use Firefox
self._journalist_logs_in()
self._journalist_checks_messages()
self._journalist_stars_and_unstars_single_message()
> self._journalist_downloads_message()

/home/kstop/projects/securedrop/securedrop-kog/securedrop/tests/functional/test_submit_and_retrieve_file.py:21:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <tests.functional.test_submit_and_retrieve_file.TestSubmitAndRetrieveFile object at 0x7faa80da4850>

def _journalist_downloads_message(self):
self._journalist_selects_the_first_source()

submissions = self.driver.find_elements_by_css_selector(
'#submissions a')
assert 1 == len(submissions)

file_url = submissions[0].get_attribute('href')

# Downloading files with Selenium is tricky because it cannot automate
# the browser's file download dialog. We can directly request the file
# using urllib2, but we need to pass the cookies for the logged in user
# for Flask to allow this.
def cookie_string_from_selenium_cookies(cookies):
cookie_strs = []
for cookie in cookies:
cookie_str = "=".join([cookie['name'], cookie['value']]) + ';'
cookie_strs.append(cookie_str)
return ' '.join(cookie_strs)

cks = cookie_string_from_selenium_cookies(self.driver.get_cookies())
raw_content = self.return_downloaded_content(file_url, cks)

> decrypted_submission = self.gpg.decrypt(raw_content)
E       AttributeError: 'TestSubmitAndRetrieveFile' object has no attribute 'gpg'

/home/kstop/projects/securedrop/securedrop-kog/securedrop/tests/functional/journalist_navigation_steps.py:655: AttributeError
```

## Comments
It looks like in setup() in securedrop/tests/functional/functional_test.py, the init_gpg() function is only called if the tests are being run against Docker, and not if there is an instance_information.json file present. This logic probably needs to be revisited, or GPG decryption needs to be handled some other way (maybe by specifying the key to use in instance_information.json?)

**Comment by redshiftzero** [comment]
Is this still an issue @zenmonkeykstop @kushaldas ?

**Comment by zenmonkeykstop** [comment]
@redshiftzero nope, fixed in #3692
